features/admin.users.v1/components/AccountStatusFilterDropdown.tsx

```diff
@@ -0,0 +1,131 @@
+
+import Badge from "@oxygen-ui/react/Badge";
+import Box from "@oxygen-ui/react/Box";
+import Checkbox from "@oxygen-ui/react/Checkbox";
+import FormControl from "@oxygen-ui/react/FormControl";
+import ListItemText from "@oxygen-ui/react/ListItemText";
+import MenuItem from "@oxygen-ui/react/MenuItem";
+import Select from "@oxygen-ui/react/Select";
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, { FunctionComponent, ReactElement } from "react";
+
+import { USER_ACCOUNT_STATUS_FILTER_OPTIONS } from "../constants/user-management-constants";
+import { AccountStatusFilterOption } from "../models/user";
+
+import "./AccountStatusFilterDropdown.scss";
+
+/**
+ * Props for the AccountStatusFilterDropdown component.
+ */
+interface AccountStatusFilterDropdownProps extends IdentifiableComponentInterface {
+    selectedFilters: string[];
+    onSelectedFiltersChange: (selected: string[]) => void;
+    label?: string;
+}
+
+/**
+ * The AccountStatusFilterDropdown component.
+ *
+ * @param props - Props injected into the component.
+ * @returns React Element.
+ */
+const AccountStatusFilterDropdown: FunctionComponent<AccountStatusFilterDropdownProps> = ({
+    selectedFilters,
+    onSelectedFiltersChange,
+    label = "Account Status",
+    ["data-componentid"]: componentId = "account-status-filter"
+}: AccountStatusFilterDropdownProps): ReactElement => {
+
+    const handleChange = (event: SelectChangeEvent<string[]>) => {
+        const value: string[] | string = event.target.value;
+
+        onSelectedFiltersChange(typeof value === "string" ? value.split(",") : value);
+    };
+
+    /**
+     * Renders the dropdown label with selection count badge.
+     * @returns The rendered component
+     */
+    const renderLabelWithCount = (): ReactElement => (
+        <Box 
+            className="dropdown-label-container"
+            data-componentid={ `${componentId}-label` }
+        >
+            { label }
+            { selectedFilters.length > 0 && (
+                <Badge
+                    className="status-count-badge"
+                    badgeContent={ selectedFilters.length }
+                    data-testid={ `${componentId}-label-badge` }
+                />
+            ) }
+        </Box>
+    );
+
+    return (
+        <Box 
+            className="account-status-filter-dropdown"
+            data-componentid={ componentId }
+        >
+            <FormControl fullWidth>
+                <Select
+                    className="account-status-select"
+                    id="account-status-select"
+                    multiple
+                    value={ selectedFilters }
+                    onChange={ handleChange }
+                    renderValue={ renderLabelWithCount }
+                    data-testid={ `${componentId}-select` }
+                    displayEmpty
+                >
+                    { USER_ACCOUNT_STATUS_FILTER_OPTIONS
+                        .sort((
```

Comment:
what is the purpose of sorting the static options here?

Reply:
During our initial discussion we decided to move the selected ones to the top. That's why the sorting was implemented here.